docs: Add audit logs docs (#3975)
coder/coder: docs/admin/audit-logs.md

# Audit Logs

> Audit logs is an enterprise feature that allows **Admins** and **Auditors** to monitoring what is happening on their deployment.

Reviewer: ```suggestion
This is an enterprise feature that allows **Admins** and **Auditors** to monitor what is happening in their deployment.
```

## Tracked Events

> The following events are tracked by this feature:

Reviewer: ```suggestion
The following events are tracked:
```

- Create
- Update

> - Delete

Reviewer: Can we elaborate? Is it any create event? (user, workspace template)

For the following resources:

- GitSSHKey
- Template
- TemplateVersion
- Workspace
- APIKey

> - User

Reviewer: Ooooo I see it now... Hmm, I wonder if we can structure this better. Maybe something like:

This feature tracks create, update and delete events for the following resources:
- list

Author: Makes sense.

## Enabling this feature

> To enable this feature an Admin needs to purchase an enterprise license.

Reviewer: And then what? It's automatically enabled?

Maybe like 
```suggestion
This feature is autoenabled for all enterprise deployments. An Admin can purchase a license [here](link)
```

Author: Yeah, we don't have a link yet :(

Reviewer: Ah gotcha. Yeah whatever you think!
